packages/devtools_app/lib/src/timeline/flutter/timeline_controller.dart

```diff
@@ -29,81 +29,76 @@ const String timelineScreenId = 'timeline';
 class TimelineController implements DisposableController {
   TimelineController() {
     timelineService = TimelineService(this);
-    fullTimeline = FullTimeline(this);
-    frameBasedTimeline = FrameBasedTimeline(this);
-    timelines = [frameBasedTimeline, fullTimeline];
+    processor = TimelineProcessor(this);
   }
 
   final cpuProfilerController = CpuProfilerController();
 
   /// Notifies that a timeline event was selected.
-  ValueListenable get selectedTimelineEventNotifier =>
-      _selectedTimelineEventNotifier;
+  ValueListenable get selectedTimelineEvent => _selectedTimelineEventNotifier;
   final _selectedTimelineEventNotifier = ValueNotifier<TimelineEvent>(null);
 
+  /// Notifies that a timeline frame has been selected.
+  ValueListenable get selectedFrame => _selectedFrameNotifier;
```

Comment:
maybe we should turn on the no-implicit dynamic lints. I wonder how many other places we have dynamic sneaking in where we don't intend it.
Change these all to
ValueListenable<someType>